src/Symfony/Component/AssetMapper/Tests/Factory/MappedAssetFactoryTest.php

```diff
@@ -121,6 +122,35 @@ public function compile(string $content, MappedAsset $asset, AssetMapperInterfac
         $this->assertSame('7e4f24ebddd4ab2a3bcf0d89270b9f30', $asset->digest);
     }
 
+    /**
+     * @dataProvider provideHashAlgorithmsDigests
+     */
+    public function testCreateMappedAssetHashAlgorithms(?string $hashAlgorithm, string $logicalPath, string $sourcePath, string $expectedDigest)
+    {
+        $factory = $this->createFactory(null, $hashAlgorithm);
+        $asset = $factory->createMappedAsset($logicalPath, $sourcePath);
+        $this->assertSame($expectedDigest, $asset->digest);
+    }
+
+    public static function provideHashAlgorithmsDigests()
+    {
+        $logicalPath = 'subdir/file6.js';
+        $sourcePath = __DIR__.'/../Fixtures/dir2/subdir/file6.js';
```

Comment:
If `$logicalPath` and `$sourcePath` never change, they're nothing but noise. I would not include them in the data provider.

Reply:
I hesitated :) Updated, thanks!